telescope/external.py

```diff
@@ -200,38 +200,35 @@ def monitor_query_queue(self, job_id, job_metadata, query_object = None, callbac
 
     if self.project_id is not None:
       started_checking = datetime.datetime.utcnow()
-
-      notification_identifier = "{metric}, {site}, {client_provider}, {date}, {duration}".format(**job_metadata)
-      self.logger.info('Queued request for {notification_identifier}, received job id: {job_id}'.format(
-          notification_identifier = notification_identifier, job_id = job_id))
-
+      
+      identifier_string = ', '.join(filter(None, job_metadata.values()))
+    
+      self.logger.info(('Queued request for ' + identifier_string  + ', received job id: {job_id}').format(
+          job_id = job_id))
+      
       while True:
         try:
           job_collection = query_object.authenticated_service.jobs()
           job_collection_state = job_collection.get(projectId = self.project_id, jobId = job_id).execute()
         except (SSLError, Exception, AttributeError, HttpError, httplib2.ServerNotFoundError) as caught_error:
           self.logger.warn(('Encountered error ({caught_error}) monitoring ' +
-                            'for {notification_identifier}, could be temporary, ' +
-                            'not bailing out.').format(caught_error = caught_error,
-                                                      notification_identifier = notification_identifier))
+                            'for ' + identifier_string  + ', could be temporary, ' +
+                            'not bailing out.').format(caught_error = caught_error))
```

Comment:
Seems weird to do concatenation in the middle of a string we're already calling `format` on. Can we keep it as a format param?